src/Utility.ts

```diff
@@ -231,3 +231,7 @@ export function memoize<Args extends unknown[], Ret> (fn: (...args: Args) => Ret
     return ret
   }
 }
+
+export const validateNonnegativeInteger = (n: number | string): boolean => {
+  return Number.isFinite(n) && !Number.isNaN(n) && Number.isInteger(n)
+}
```

Comment:
we should probably just use zod for this.

z.number().int().safe().safeParse(/*  value */).success

Reply:
.safe() restricts integers to only within javascript Floating Point Precision, but ideally this would be used to allow any integer up to ~1e308. However this seems to not be used in this case, so I may go ahead with this suggestion anyway